projects/jupyter-server-ydoc/jupyter_server_ydoc/handlers.py

```
@@ -459,3 +463,125 @@ async def put(self, path):
         )
         self.set_status(201)
         return self.finish(data)
+
+
+class TimelineHandler(APIHandler):
+    def initialize(
+        self, ystore_class: type[BaseYStore], ywebsocket_server: JupyterWebsocketServer
+    ) -> None:
+        self.ystore_class = ystore_class
+        self.ywebsocket_server = ywebsocket_server
+
+    async def get(self, path: str) -> None:
+        idx = uuid4().hex
+        file_id_manager = self.settings["file_id_manager"]
+        file_id = file_id_manager.get_id(path)
+
+        format = str(self.request.query_arguments.get("format")[0].decode("utf-8"))
+        content_type = str(self.request.query_arguments.get("type")[0].decode("utf-8"))
+        encoded_path = encode_file_path(format, content_type, file_id)
+        try:
+            room_id = room_id_from_encoded_path(encoded_path)
+            room: YRoom = await self.ywebsocket_server.get_room(room_id)
+            fork_ydoc = Doc()
+
+            ydoc_factory = YDOCS.get(content_type)
+            if ydoc_factory is not None:
```

The handler should return an error when there is no document factory.